clustered/common/src/main/java/org/ehcache/clustered/common/messages/EhcacheCodec.java

```diff
@@ -34,30 +34,28 @@
 
   @Override
   public byte[] encodeMessage(EhcacheEntityMessage message) {
-    if (message.getType() == EhcacheEntityMessage.Type.LIFECYCLE_OP) {
-      return LifeCycleOpCodec.encode((LifecycleMessage) message);
-    } else {
-      return ServerStoreOpCodec.encode((ServerStoreOpMessage) message);
-    }
+    return message.encode();
```

Comment:
Think of the future when multiple versions of the server have been released. When that happens, we'll have to support multiple versions of the client connecting on the server to allow hot upgrades.